assets/model_monitoring/components/src/model_data_collector_preprocessor/mdc_preprocessor_helper.py

```diff
@@ -0,0 +1,210 @@
+
+    sub_id = sub_id or os.environ.get("AZUREML_ARM_SUBSCRIPTION", None)
+    rg_name = rg_name or os.environ.get("AZUREML_ARM_RESOURCEGROUP", None)
+    ws_name = ws_name or os.environ.get("AZUREML_ARM_WORKSPACE_NAME", None)
+
+    return f"azureml://subscriptions/{sub_id}/resourcegroups/{rg_name}/workspaces/{ws_name}/datastores" \
+           f"/{datastore}/paths/{path}"
+
+
+def get_hdfs_path_and_container_client(uri_folder_path: str, ws: Workspace = None) \
+        -> Tuple[str, FileSystemClient | ContainerClient]:
+    """Get HDFS path and service client from url_folder_path."""
+    url = urlparse(uri_folder_path)
+    # TODO sovereign endpoint
+    if url.scheme in ["https", "http"]:
+        pattern = r"(?P<scheme>http|https)://(?P<account_name>[^\.]+).(?P<store_type>blob|dfs).core.windows.net/" \
+                  r"(?P<container>[^/]+)/(?P<path>.+)"
+        matches = re.match(pattern, uri_folder_path)
+        if not matches:
+            raise ValueError(f"Unsupported uri as uri_folder: {uri_folder_path}")
+        # use abfss to access both blob and dfs, to workaround the append block issue
+        scheme = "abfss" if matches.group("scheme") == "https" else "abfs"
+        store_type = "dfs"
+        account_name = matches.group("account_name")
+        container = matches.group("container")
+        path = matches.group("path")
+        return f"{scheme}://{container}@{account_name}.{store_type}.core.windows.net/{path}", None
+    elif url.scheme in ["wasbs", "wasb", "abfss", "abfs"]:
+        pattern = r"(?P<scheme>wasbs|abfss|wasb|abfs)://(?P<container>[^@]+)@(?P<account_name>[^\.]+).(?P<store_type>blob|dfs).core.windows.net/(?P<path>.+)"  # noqa
+        matches = re.match(pattern, uri_folder_path)
+        if not matches:
+            raise ValueError(f"Unsupported uri as uri_folder: {uri_folder_path}")
+        scheme = "abfss" if matches.group("scheme") in ["wasbs", "abfss"] else "abfs"
+        store_type = "dfs"
+        account_name = matches.group("account_name")
+        container = matches.group("container")
+        path = matches.group("path")
+        return f"{scheme}://{container}@{account_name}.{store_type}.core.windows.net/{path}", None
+    elif url.scheme == "azureml":
+        if ':' in url.path:  # azureml asset path
+            # asset path should be translated to azureml or hdfs path in service, should not reach here
+            raise ValueError("AzureML asset path is not supported as uri_folder.")
+        else:  # azureml long or short form
+            datastore_name, path = _get_datastore_and_path_from_azureml_path(uri_folder_path)
+            ws = ws or Run.get_context().experiment.workspace
+            datastore = ws.datastores.get(datastore_name)
+            datastore_type = datastore.datastore_type
+            if datastore_type not in ["AzureBlob", "AzureDataLakeGen2"]:
+                raise ValueError(f"Only Azure Blob and Azure Data Lake Gen2 are supported, but got {datastore.type}.")
```

Comment:
Can we define a validation user error type?

Reply:
ValueError is the built-in error for invalid user input, do we really want to define our own validation error?

Reply:
So in the dashboard, we filter out all the value error?